src/self-managed-debug/src/docker_resource_dumper.rs

```diff
@@ -0,0 +1,158 @@
+                        .await;
+
+                    match output {
+                        Ok(output) if output.status.success() => {
+                            RetryResult::Ok((output.stdout, output.stderr))
+                        }
+                        Ok(output) => {
+                            let err_msg = format!(
+                                "Docker command failed: {:#}. Retrying...",
+                                String::from_utf8_lossy(&output.stderr)
+                            );
+                            error!("{}", err_msg);
+                            RetryResult::RetryableErr(anyhow::anyhow!(err_msg))
+                        }
+                        Err(err) => {
+                            let err_msg = format!("Failed to execute Docker command: {:#}", err);
+                            error!("{}", err_msg);
+                            RetryResult::RetryableErr(anyhow::anyhow!(err_msg))
+                        }
+                    }
+                }
+            })
+            .await
+    }
+
+    async fn dump_logs(&self) -> Result<(), anyhow::Error> {
+        let file_path = format_docker_resource_path(self.context.start_time, &self.container_id);
+
+        let (stdout, stderr) = self
+            .execute_docker_command(&["logs".to_string(), self.container_id.to_string()])
+            .await?;
+
+        write_output(stdout, "logs-stdout.txt", &file_path)?;
+        write_output(stderr, "logs-stderr.txt", &file_path)?;
+
+        Ok(())
+    }
+
+    async fn dump_inspect(&self) -> Result<(), anyhow::Error> {
+        let file_path = format_docker_resource_path(self.context.start_time, &self.container_id);
+
+        let (stdout, _) = self
+            .execute_docker_command(&["inspect".to_string(), self.container_id.to_string()])
+            .await?;
+
+        write_output(stdout, "inspect.txt", &file_path)?;
+
+        Ok(())
+    }
+
+    async fn dump_stats(&self) -> Result<(), anyhow::Error> {
+        let file_path = format_docker_resource_path(self.context.start_time, &self.container_id);
+
+        let (stdout, _) = self
+            .execute_docker_command(&[
+                "stats".to_string(),
+                "--no-stream".to_string(),
+                self.container_id.to_string(),
+            ])
+            .await?;
+
+        write_output(stdout, "stats.txt", &file_path)?;
+
+        Ok(())
+    }
+}
+
+/// Dump all Docker resources.
+pub async fn dump_all_docker_resources(context: &Context) -> () {
+    let container_id = context
+        .args
+        .docker_container_id
+        .clone()
+        .expect("No container ID provided");
+    let dumper = DockerResourceDumper::new(context, container_id);
+
+    let _ = dumper.dump_logs().await;
+    let _ = dumper.dump_inspect().await;
+    let _ = dumper.dump_stats().await;
+}
+
+/// Helper closure to write output to file
+fn write_output(
+    output: Vec<u8>,
+    file_name: &str,
+    file_path: &PathBuf,
```

Comment:
Extra nitpick, when I see the name `file_path`, I assume it's a path to a file, not a directory. Perhaps `directory_path` would be a better name.

The order of these args is also a bit weird. The file name is appended to the path, so it would read slightly better to come after the directory path.